java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -52,14 +52,6 @@ public String toString() {
   }
 
   @Override
-  public boolean equals(Object o) {
```

Comment:
I am not sure why this code was wrong -- @elahrvivaz can you verify that this doesn't break your downstream application? I will hold off on merging for now